Scripts/Pointer/StraightPointer.cs

```diff
@@ -0,0 +1,176 @@
+﻿namespace VRTK.Core.Pointer
+{
+    using UnityEngine;
+    using VRTK.Core.Utility;
+
+    /// <summary>
+    /// The StraightPointer is a pointer implementation that casts a direct line from the point of origin until it collides with a valid target or reaches it's maximum length.
+    /// </summary>
+    public class StraightPointer : BasePointer
+    {
+        [Header("Pointer Settings")]
+
+        [Tooltip("The maximum length to cast the pointer.")]
+        public float maximumLength = 100f;
+        [Tooltip("The GameObject to represent the pointer's tracer when it is colliding with a valid object.")]
+        public GameObject validTracer;
+        [Tooltip("The GameObject to represent the pointer's cursor when it is colliding with a valid object.")]
+        public GameObject validCursor;
+        [Tooltip("The GameObject to represent the pointer's tracer when it is colliding with an invalid object or not colliding at all.")]
+        public GameObject invalidTracer;
+        [Tooltip("The GameObject to represent the pointer's cursor when it is colliding with an invalid object or not colliding at all.")]
+        public GameObject invalidCursor;
+        [Tooltip("Determines when the pointer tracer GameObject should be seen.")]
+        public VisibilityType tracerVisibility = VisibilityType.OnWhenActive;
+        [Tooltip("Determines when the pointer cursor GameObject should be seen.")]
+        public VisibilityType cursorVisibility = VisibilityType.OnWhenActive;
+        [Tooltip("Determines if the pointer should be automatically activated when the script is enabled.")]
+        public bool activateOnEnable;

```

Comment:
See review comment on `BasePointer`: Maybe we can merge "active" and `MonoBehaviour`'s `enabled`?

Reply:
this is different again. this is if this is ticked then call the `Activate()` method when the script becomes enabled.